index.js

```diff
@@ -0,0 +1,680 @@
+  //. | [Apply][]       |     ✔︎     |     ✔︎     |           |           |
+  //. | [Applicative][] |     ✔︎     |           |           |           |
+  //. | [Chain][]       |     ✔︎     |           |           |           |
+  //. | [ChainRec][]    |           |           |           |           |
+  //. | [Monad][]       |           |           |           |           |
+  //. | [Foldable][]    |     ✔︎     |           |           |           |
+  //. | [Traversable][] |     ✔︎     |           |           |           |
+  //. | [Extend][]      |           |           |           |           |
+  //. | [Comonad][]     |           |           |           |           |
+
+  var methods = {
+
+    //  Array#toString :: Array a ~> String
+    'Array#toString': function() {
+      return '[' + this.map(Z.toString).join(', ') + ']';
+    },
+
+    //  Array#equals :: Array a ~> Array a -> Boolean
+    'Array#fantasy-land/equals': function(other) {
+      return other.length === this.length &&
+             this.every(function(x, idx) { return equals(other[idx], x); });
+    },
+
+    //  Array#concat :: Array a ~> Array a -> Array a
+    'Array#fantasy-land/concat': function(xs) { return this.concat(xs); },
+
+    //  Array#empty :: Array a ~> Array a
+    'Array#fantasy-land/empty': function() { return []; },
+
+    //  Array#map :: Array a ~> (a -> b) -> Array b
+    'Array#fantasy-land/map': function(f) {
+      return this.map(function(x) { return f(x); });
+    },
+
+    //  Array#ap :: Array (a -> b) ~> Array a -> Array b
+    'Array#fantasy-land/ap': function(xs) {
+      var result = [];
+      for (var idx = 0; idx < this.length; idx += 1) {
+        for (var idx2 = 0; idx2 < xs.length; idx2 += 1) {
+          result.push(this[idx](xs[idx2]));
+        }
+      }
+      return result;
+    },
+
+    //  Array#of :: Array a ~> b -> Array b
+    //
+    //  This method is never actually used, but it must be defined in order
+    //  for `Applicative.test([])` to evaluate `true` in environments without
+    //  [`Array.of`][Array.of].
+    'Array#fantasy-land/of': /* istanbul ignore next */ function(x) {
+      return [x];
+    },
+
+    //  Array#chain :: Array a ~> (a -> Array b) -> Array b
+    'Array#fantasy-land/chain': function(f) {
+      var result = [];
+      for (var idx = 0; idx < this.length; idx += 1) {
+        push_.apply(result, f(this[idx]));
+      }
+      return result;
+    },
+
+    //  Array#reduce :: Array a ~> ((b, a) -> b, b) -> b
+    'Array#fantasy-land/reduce': function(f, initial) {
+      return this.reduce(function(acc, x) { return f(acc, x); }, initial);
+    },
+
+    //  Array#traverse :: Applicative f => Array a ~> ((a -> f b), (c -> f c)) -> f (Array b)
+    'Array#fantasy-land/traverse': function(f, of) {
+      //  prepend :: a -> Array a -> Array a
+      var prepend = function(x) {
+        return function(xs) {
+          return [x].concat(xs);
+        };
+      };
```

Comment:
Good idea. I've just made your suggested change. :)

Thanks for taking the time to review this pull request. I'm pleased you like the approach to dispatching.